src/python/BuildingControlsSimulator/DataClients/WeatherChannel.py

```diff
@@ -703,3 +734,303 @@ def to_epw(
             epw_data[self.epw_columns].to_csv(f, header=False, index=False)
 
         return fpath
+
+
+    #BCS - For this to work you must:
+    # 1) install h5pyd:   pip install --user h5pyd
+    # 2) Get your own API key, visit https://developer.nrel.gov/signup/
+    # 3) configure HSDS:  Add the following contents to a configuration file at ~/.hscfg:
+        # #HDFCloud configuration file
+        # hs_endpoint = https://developer.nrel.gov/api/hsds
+        # hs_username = None
+        # hs_password = None
+        # hs_api_key = <your api key here>
+
+    def get_nsrdb(self,lat,long):
+        
+        # Unlike the gridded WTK data the NSRDB is provided as sparse time-series dataset.
+        # The quickest way to find the nearest site it using a KDtree
+        def nearest_site(tree, lat_coord, lon_coord):
+            lat_lon = np.array([lat_coord, lon_coord])
+            dist, pos = tree.query(lat_lon)
+            return pos
+        
+        # Open the desired year of nsrdb data
+        # server endpoint, username, password is found via a config file
+        f = h5pyd.File("/nrel/nsrdb/v3/nsrdb_2019.h5", 'r')
```

Comment:
Are there other years available? We should use the sim_config start and end times to determine which year to use.

Reply:
possible! I haven't tried since we are currently focussing on the 2019 set

Reply:
I know 2020 wasn't available when last checked.  but this could work for prior years.  I'll do a quick dig.

Reply:
interesting. nsrdb data goes back to as early as 2000!  I didn't bother digging further back.  2020 still not available.

Reply:
done.